scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala

```diff
@@ -298,7 +305,92 @@ object node {
   // shape of this data should probably change, currently we leave it for backward compatibility
   object FragmentInputDefinition {
 
-    @JsonCodec case class FragmentParameter(name: String, typ: FragmentClazzRef)
+    @JsonCodec case class FixedExpressionValue(expression: String, label: String)
+
+    object FragmentParameterInputMode extends Enumeration {
+      type FragmentParameterInputMode = Value
+
+      implicit val typeEncoder: Encoder[FragmentParameterInputMode.Value] =
+        Encoder.encodeEnumeration(FragmentParameterInputMode)
+      implicit val typeDecoder: Decoder[FragmentParameterInputMode.Value] =
+        Decoder.decodeEnumeration(FragmentParameterInputMode)
+
+      val InputModeAny: Value                = Value("InputModeAny")
+      val InputModeAnyWithSuggestions: Value = Value("InputModeAnyWithSuggestions")
+      val InputModeFixedList: Value          = Value("InputModeFixedList")
+    }
+
+    object FragmentParameter {
+
+      def apply(name: String, typ: FragmentClazzRef): FragmentParameter = {
+        FragmentParameterNoFixedValues(name, typ, required = false, None, None, InputModeAny)
+      }
+
+    }
+
+    sealed trait FragmentParameter {
```

Comment:
this trait, its implementation and jsoncodecs might seem like overkill, but I think it' the most sensible option for the near future, when a third case (FragmentParameterFixedValuesPreset) is added